code/datums/components/infective.dm

```diff
@@ -34,6 +37,11 @@
 		eater.ForceContractDisease(V)
 	try_infect(feeder, BODY_ZONE_L_ARM)
 
+/datum/component/infective/proc/try_infect_drink(datum/source, mob/living/drinker, mob/living/feeder)
+	for(var/disease in diseases)
+		drinker.ForceContractDisease(disease)
+	try_infect(feeder, BODY_ZONE_L_ARM)
```

Comment:
```suggestion
	var/appendage_zone
	switch(held_items.Find(source))
		if(0)
			appendage_zone = pick(BODY_ZONE_L_ARM, BODY_ZONE_R_ARM)
		if(1)
			appendage_zone = BODY_ZONE_L_ARM
		if(2)
			appendage_zone = BODY_ZONE_R_ARM
	try_infect(feeder, appendage_zone)
```

Reply:
No this makes sense, it forces the drinker to contract every disease that the drink has regardless. the one thats feeding the drinker has a chance of getting infected

Reply:
Ah, you are right, then it's on the hands. The `switch` version should take care of which hand is the correct one.
If you want to support multiple hands you could also do something like this:
```suggestion
	var/appendage_zone = held_items.Find(source)
	appendage_zone = appendage_zone == 0 ? pick(BODY_ZONE_L_ARM, BODY_ZONE_R_ARM)  : appendage_zone % 2 ? BODY_ZONE_R_ARM : BODY_ZONE_L_ARM
	try_infect(feeder, appendage_zone)
```